File: php/src/BowlingGame.php
<?php


namespace Armakuni\BadBowlingKata;


class BowlingGame
{
    /**
     * @var array
     */
    private $rolls;

    /**
     * BowlingGame constructor.
     */
    public function __construct()
    {
        $this->rolls = [];
    }

    public function roll(int $pins): void
    {
        $this->rolls[] = $pins;
    }

    public function score(): int
    {
        $score = 0;
        $nextScoredRoll = 0;

        foreach (range(1, 10) as $frame) {
            $frameScore = 0;
            if ($this->rolls[$nextScoredRoll] == 10) {
                $frameScore += $this->rolls[$nextScoredRoll];
                $frameScore += $this->rolls[$nextScoredRoll + 1];
                $frameScore += $this->rolls[$nextScoredRoll + 1];
                $nextScoredRoll += 1;
            } elseif (($this->rolls[$nextScoredRoll] + $this->rolls[$nextScoredRoll + 1]) == 10) {
                $frameScore += $this->rolls[$nextScoredRoll];
                $frameScore += $this->rolls[$nextScoredRoll + 1];
                $frameScore += $this->rolls[$nextScoredRoll + 2];
                $nextScoredRoll += 1;
                $nextScoredRoll += 1;
            } else {
                $frameScore += $this->rolls[$nextScoredRoll];
                $frameScore += $this->rolls[$nextScoredRoll + 1];
                $nextScoredRoll += 1;
                $nextScoredRoll += 1;
            }

            $score += $frameScore;
        }

        return $score;
    }
}

File: python/bowling_game.py
class BowlingGame:
    _rolls: list[int]

    def __init__(self) -> None:
        self._rolls = []

    def roll(self, pins: int) -> None:
        self._rolls.append(pins)

    def score(self) -> int:
        score = 0
        next_scored_roll = 0

        for _ in range(10):
            frame_score = 0

            if self._rolls[next_scored_roll] == 10:
                frame_score += self._rolls[next_scored_roll]
                frame_score += self._rolls[next_scored_roll + 1]
                frame_score += self._rolls[next_scored_roll + 1]
                next_scored_roll += 1
            elif (
                self._rolls[next_scored_roll] + self._rolls[next_scored_roll + 1] == 10
            ):
                frame_score += self._rolls[next_scored_roll]
                frame_score += self._rolls[next_scored_roll + 1]
                frame_score += self._rolls[next_scored_roll + 2]
                next_scored_roll += 1
                next_scored_roll += 1
            else:
                frame_score += self._rolls[next_scored_roll]
                frame_score += self._rolls[next_scored_roll + 1]
                next_scored_roll += 1
                next_scored_roll += 1

            score += frame_score

        return score
